src/zope/app/testing/dochttp.py

```diff
@@ -116,18 +120,22 @@ def dochttp(args=sys.argv[1:], default=None):
                     break
             else:
                 try:
-                    output_test(request, response, options.clean_redirects)
-                except IOError, e:
+                    output_test(request, response, options.clean_redirects, output_fp)
+                except IOError as e:
                     if e.errno == errno.EPIPE:
                         return
                     raise
 
 
-def output_test(request, response, clean_redirects=False):
-    print
-    print
-    print '  >>> print http(r"""'
-    print '  ...', '\n  ... '.join(request.lines())+'""")'
+def output_test(request, response, clean_redirects=False, output_fp=None):
+    if output_fp is None:
+        output_fp = sys.stdout
+    request_lines = [x.decode("latin-1") if not isinstance(x, str) else x
+                     for x in request.lines()]
+    print(file=output_fp)
+    print(file=output_fp)
+    print('  >>> print(http(r"""', file=output_fp)
+    print('  ...', '\n  ... '.join(request_lines) + '"""))', file=output_fp)
```

Comment:
TBH I sincerely hope nobody is using these old-old-old-pre-zope.testbrowser-style tests that put raw HTTP requests and responses in the doctest.

The idea of recording live browser sessions with [tcpwatch](https://pypi.python.org/pypi/tcpwatch) and converting those to functional tests was a nice one, but didn't work out.

I think this script can die, like zope.app.testing.testbrowser.

Reply:
For better or worse, there are a large number of "raw HTTP" tests hanging around in the various zope.app packages. 

I always wondered who had the patience to sit down and write those things out. Now I know.

Reply:
The way I remember, I was mostly copying and pasting existing ones, then changing query strings and response texts, because setting up tcpwatch was too much hassle to be worth the effort. ;)